### PR TITLE
added tests and fleshed out connect4 package

### PR DESCRIPTION
**Pull request by chriscole-whs**

### CODE REVIEW

File: internal/connect4/connect4.go [hunk 1]
```diff
@@ -1,27 +1,43 @@
 package connect4
 
+const (
+	RedPlayer = "red"
+	BlackPlayer = "black"
+	DrawResult = "draw"
+
+	ErrIllegalColumn = "illegal column"
```
Comment (scags9876) [inline internal/connect4/connect4.go]: these should be actual error types

File: main.go [hunk 2]
```diff
@@ -5,22 +5,167 @@ import (
...
+func testGame() {
 	game := connect4.NewGame()
 
-	testGame(game)
+	var succeeded, failed int
+
+	// Illegal move test cases --------------------------------------------------
+
+	// Out of bounds error (0)
+	game, error := runTestCase([]int{0}, game)
```
Comment (scags9876) [inline main.go]: variable name should be `err`

**Review by scags9876 — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  Game did not report that Black won the game
+
+
+Test results:
+  Passed 1 out of 11 tests
+```
+
+### Submission Instructions
+Clone this repository, and make a branch of your own.
+Once you have an implementation that passes the tests in main.go, make a pull request
```
Comment (scags9876) [inline README.md]: maybe put a line in there that reads something like `Come up with a data structure for the connect4.Game struct and implement the functions in the connect4 package`

File: internal/connect4/connect4.go [hunk 4]
```diff
@@ -1,27 +1,45 @@
...
 
 // IsOver indicates if the game has been won or if it is a draw (no moves remaining)
 func (g *Game) IsOver() bool {
-
+	return false
 }
 
-// Result should a string describing which player won, or if it was a draw
+// Result should return one of the constants defined in this file
+// describing which player won, if it was a draw, or the empty string if the game is ongoing
```
Comment (scags9876) [inline internal/connect4/connect4.go]: Result returns a single word, either "draw", or "red" or "black", or an empty string if there is not yet a result

File: README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+against this repo.
```
Comment (scags9876) [inline README.md]: maybe we should ask them to create a gist with their connect4 package instead of a PR.. might be easier to clean up

Comment (chriscole-whs) [reply README.md]: Good point.

**Review by joncarey-whitehatsec — APPROVED** [review]
Looks good!

**Review by scags9876 — APPROVED** [review]
lgtm

**Review by scags9876 — APPROVED** [review]

**Pull request merged**